### ISSUE TITLE
helper for datetime conversions

### ISSUE THREAD

**Issue by fdncred** [nushell/engine-q#869]
Something like this would be helpful as a `into datetime --list` type functionality. I know i can never remember these things.
https://www.strfti.me/
https://strftime.org/

**Comment by voidash** [comment]
Hello, this is going to be my first contribution to this project. When i peeked into the code, i found out that the implementation is using `std::chrono` so  i built a cheatsheet from [https://docs.rs/chrono/0.3.1/chrono/format/strftime/index.html](https://docs.rs/chrono/0.3.1/chrono/format/strftime/index.html) 
Here is the help message for list

```
 -l, --list
      lists strftime cheatsheet
```
I used `Value::Record` to show the output. Here is the sample output
```
/home/cdjk/ >  '2020-09-04_16:39:18+0000' | into datetime --list                                       
╭──────┬─────────────────────────────────────────────────────────────────────────────────────────╮
│ code │ Description                                                                             │
│ %F   │ Year-month-day format. Same to %Y                                                       │
│ %Y   │ The full proleptic Gregorian year, zero-padded to 4 digits(Example: 2001).              │
│ %C   │ The proleptic Gregorian year divided by 100, zero-padded to 2 digits. (Example: 20)     │
│ %y   │ The proleptic Gregorian year modulo 100, zero-padded to 2 digits. (Example: 01)         │
│ %m   │ Month number (01--12), zero-padded to 2 digits.(Example: 07)                            │
│ %b   │ Abbreviated month name. Always 3 letters.(Example: Jul)                                 │
│ %B   │ Full month name. Also accepts corresponding abbreviation in parsing.(Example: July)     │
│ %h   │ Same to %b                                                                              │
│ %d   │ Day number (01--31), zero-padded to 2 digits.(Example: 08)                              │
│ %e   │ Same to %d                                                                              │
│ %a   │ Abbreviated weekday name. Always 3 letters.(Example: Sun)                               │
│ %A   │ Full weekday name. Also accepts corresponding abbreviation in parsing.(Example: Sunday) │
│ %w   │ Sunday = 0, Monday = 1, ..., Saturday = 6.(Example: 0)                                  │
│ %u   │ Monday = 1, Tuesday = 2, ..., Sunday = 7. (ISO 8601)(Example: 7)                        │
```

**Comment by fdncred** [comment]
@voidash yes, good work. i love this. i left some comments on the pr. thank you very much!!!!

**Comment by fdncred** [comment]
closed due to #883 yay!

**Comment by fdncred** [comment]
> For discoverability, it might be worth providing `date format --list` as well with the same table

good idea. we'd accept a pr for that. :)
